src/main/java/com/example/javawebcourse/featuretoggle/aspect/FeatureToggleAspect.java

```diff
@@ -0,0 +1,38 @@
+package com.example.javawebcourse.featuretoggle.aspect;
+
+
+import com.example.javawebcourse.featuretoggle.FeatureToggle;
+import com.example.javawebcourse.featuretoggle.FeatureToggleService;
+import com.example.javawebcourse.featuretoggle.FeatureToggles;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import com.example.javawebcourse.featuretoggle.exception.FeatureNotAvailableException;
+
+@Slf4j
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class FeatureToggleAspect {
+
+    private final FeatureToggleService featureToggleService;
+
+    @Around(value = "@annotation(featureToggle)")
```

Comment:
```suggestion
    @Before(value = "@annotation(featureToggle)")
```

@Around is the most powerful pointcut of aspectj, which is why it is recommended to use only when you need full control over method execution. In your case, @Before is much better suit since you need to intercept method BEFORE its execution and dont care what happens afterwards. Generally, backend tries to minimize amount of information and controll each component of system has so it results in looser coupling and better mentainability. If interested, see reference: [Declaring a pointcut](https://docs.spring.io/spring-framework/reference/core/aop/ataspectj/advice.html)

Quote from source:

> Always use the least powerful form of advice that meets your requirements.
> 
> For example, do not use around advice if before advice is sufficient for your needs.